### ISSUE TITLE
Add YAML bibliography writer

### ISSUE THREAD

**Issue by hseg** [jgm/pandoc#9617]
**Describe your proposed improvement and the problem it solves.**

Given that pandoc expects in-frontmatter bibliography in YAML, it would make sense to have a `cslyaml` writer in addition to the `csljson` one

**Describe alternatives you've considered.**

Given the suggestion in #8801, I've looked around for a YAML CSL style, but haven't found anything. Of course, there's always my current workaround of
```bash
pandoc -f bibtex -t csljson \
  | yq -y "walk(if type==\"object\" then with_entries(.key|=ascii_downcase) else . end) | map(pick(.id, .[keys[]]))"
```
but I'd prefer not to have to do this kind of postprocessing.

**Comment by jgm** [comment]
`pandoc -f bibtex -s -t markdown`
should do the job!

**Comment by hseg** [comment]
That it does! Mod the irrelevant annoyance that it also generates a `nocite` key and surrounding lines, but that can be coped with.
This could stand to be documented more clearly, but other than that this can be closed

**Comment by jgm** [comment]
One issue is that, as far as I know, CSL YAML isn't specified anywhere.

`-t markdown` will put the content in Markdown, and that probably isn't wanted.  But I don't know if it's stated anywhere what the format should be in CSL YAML?  If it should be exactly as in CSL JSON, then perhaps a new writer is called for.

**Comment by jgm** [comment]
@bdarcus may be able to offer some illumination on the status of "CSL YAML."

**Comment by bdarcus** [comment]
> @bdarcus may be able to offer some illumination on the status of "CSL YAML."

Unfortunately, its status is still really uncertain, along with everything else in the CSL 1.1 branch ([this thread](https://discourse.citationstyles.org/t/upcoming-csl-meetup-context/1767) covers the questions we discussed in 2022 for those curious).

But I can say this:

These days, one can use JSON Schemas to validate YAML files.

So the schema there is designed for both:

https://github.com/citation-style-language/schema/blob/v1.1/schemas/input/csl-data.json

It's just we have no process by which we actually approve and release that and the other changes. So in practice, CSL is frozen, and along with it CSL YAML/JSON.

One could use the current JSON Schema for validating YAML, but it's less than ideal I think.

PS - In my experimental radical rewrite of CSL, I also rewrote the input model; it borrows a few ideas from the 1.1 version ([YAML example](https://github.com/bdarcus/csln/blob/main/processor/examples/chicago.bib.yaml)).

**Comment by hseg** [comment]
Indeed, that's been a point of confusion for me, and one reason I've stuck to pretty-printing the CSL JSON.

BTW, isn't `-f bibtex -st markdown` supposed to create a YAML metadata block? In particular, isn't the last line supposed to be `...`, not `---`?

**Comment by jgm** [comment]
Somewhere along the way, I got convinced to use `---` instead of `...` for the final delimiter for YAML metadata -- I think the reason was increased compatibility with other md implementations, but I can't remember.

**Comment by jgm** [comment]
It would be fairly trivial to modify the existing CslJson writer and reader to add readCslYaml and writeCslYaml (where these would *just* use YAML instead of JSON, everything else being the same).  IF there is actually a demand for this...

**Comment by bpj** [comment]
@bdarcus I even write my JSON schemas in YAML. I use a Perl script using [YAML][]/[JSON][]/[JSON Schema][] modules available on CPAN to load both data and schemas and then validate. In fact it is possible to validate any data representable as JSON once you have it loaded into a Perl data structure!

[JSON Schema]: https://metacpan.org/dist/JSON-Validator
[YAML]: https://metacpan.org/dist/YAML-PP
[JSON]: https://metacpan.org/dist/JSON

**Comment by bdarcus** [comment]
> @bdarcus I even write my JSON schemas in YAML.

I started to do that with an experiment in that branch.

https://github.com/citation-style-language/schema/blob/v1.1/schemas/input/csl-rich-text.yaml

Definitely nicer.

But the CSLN schemas are generated from the Rust models, and I use serde to read the JSON and YAML.
